### PR TITLE
[iOS] Implement Sponsor screen

### LINKED ISSUE #494: iOS App sponsor view
## Kind (Required)
- Proposal / Discussion

 ## Overview (Required)
- A view that shows sponsors.

 ## Links	
- [Design](https://www.figma.com/file/4r9becvhDy3GfXaXex8E8d/App?node-id=22%3A507)

### PR DESCRIPTION
**Pull request by yaslab**
## Issue
- close #494

## Overview (Required)
- Implemented Sponsor screen.
- I imitated most of the code on Session screen.

## Links
- N/A

## Screenshot
- ~TODO: Gold sponsor `Google Developers` is not showing 👀~
    - Solved: Because I was using a debug API. There was no problem when using a production API.

### CODE REVIEW

**Comment by yaslab** [comment]
Please review someone. 🙏

**Comment by roana0229** [comment]
👀

File: ios-base/DroidKaigi 2020/Sponsor/Views/SponsorViewController.swift [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            )
+        }
+    }
+
+    private lazy var loadingIndicatorView: UIActivityIndicatorView = {
+        let indicator = UIActivityIndicatorView(style: .gray)
+        return indicator
+    }()
+
+    private lazy var retryButton: UIButton = {
```
Comment (roana0229) [inline ios-base/DroidKaigi 2020/Sponsor/Views/SponsorViewController.swift]: In my opinion.
I think it is better to put "view" from "xib" than "code".
What do you think about put from `xib`, and change `isHidden = false` if occur error ?

File: ios-base/DroidKaigi 2020/Sidebar/Views/SidebarViewController.swift [hunk 2]
```diff
@@ -51,7 +53,9 @@ final class SidebarViewController: UITableViewController {
         case .map:
             break
         case .sponsor:
-            break
+            let vc = SponsorViewController()
+            rootViewController?.setViewControllers([vc], animated: false)
```
Comment (roana0229) [inline ios-base/DroidKaigi 2020/Sidebar/Views/SidebarViewController.swift]: We talk about this with @takahirom  and decided.
It show `drawer` on only session list screen.
Could you please change to use `pushViewController ` ? 
Like [this PR](https://github.com/DroidKaigi/conference-app-2020/blob/ef720095e8593eab2736a8e16f91dd77bb7c9749/ios-base/DroidKaigi%202020/Sidebar/Views/SidebarViewController.swift#L51) changes 🙏

File: data/repository/src/commonMain/kotlin/io/github/droidkaigi/confsched2020/data/repository/SponsorListResponseToModelMapper.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        .groupBy { sponsorResponse -> sponsorResponse.plan }
+        .mapNotNull { (plan, sponsors) ->
+            val category = SponsorCategory.Category.from(plan)
+                ?: return@mapNotNull null
+            SponsorCategory(
+                category,
+                sponsors.map { sponsor ->
+                    Sponsor(
+                        id = sponsor.id,
+                        plan = plan,
```
[the remark this answers is not in the document]
Comment (yaslab) [reply data/repository/src/commonMain/kotlin/io/github/droidkaigi/confsched2020/data/repository/SponsorListResponseToModelMapper.kt]: I will fix it!

File: ios-base/DroidKaigi 2020/Sponsor/Views/Cells/SponsorDividerView.swift [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+    @IBOutlet private var dividerView: UIView!
+
+    var isLastSection: Bool = false {
+        didSet {
+            if isLastSection {
+                dividerView.backgroundColor = .clear
+            } else {
+                if #available(iOS 13.0, *) {
+                    dividerView.backgroundColor = .systemFill
```
[the remark this answers is not in the document]
Comment (yaslab) [reply ios-base/DroidKaigi 2020/Sponsor/Views/Cells/SponsorDividerView.swift]: 😄

File: ios-base/DroidKaigi 2020/Sponsor/Views/SponsorViewDataSource.swift [hunk 5]
```diff
@@ -0,0 +1,116 @@
+import ios_combined
+import Nuke
+import RxCocoa
+import RxSwift
+import UIKit
+
+final class SponsorViewDataSource: NSObject, UICollectionViewDataSource {
+    typealias Element = [SponsorCategory]
+    var items: Element = []
```
[the remark this answers is not in the document]
Comment (yaslab) [reply ios-base/DroidKaigi 2020/Sponsor/Views/SponsorViewDataSource.swift]: I will fix it!

File: ios-base/DroidKaigi 2020/Sponsor/Views/SponsorViewController.swift [hunk 6]
```diff
@@ -0,0 +1,136 @@
+import ios_combined
+import RxCocoa
+import RxSwift
+import SafariServices
+import UIKit
+
+final class SponsorViewController: UIViewController {
+    private let disposeBag = DisposeBag()
+
+    @IBOutlet private var collectionView: UICollectionView! {
```
[the remark this answers is not in the document]
Comment (yaslab) [reply ios-base/DroidKaigi 2020/Sponsor/Views/SponsorViewController.swift]: I will fix it!

I think so too. `weak` is better for `@IBOutlet`, which the same lifetime as the view controller.

File: ios-base/DroidKaigi 2020/Sponsor/Views/SponsorViewController.swift (hunk 1 above)
Comment (yaslab) [reply ios-base/DroidKaigi 2020/Sponsor/Views/SponsorViewController.swift]: Exactry! I use xib elsewhere, It's strange that I use the code here...

I will fix it to use xib.

File: ios-base/DroidKaigi 2020/Sidebar/Views/SidebarViewController.swift (hunk 2 above)
Comment (yaslab) [reply ios-base/DroidKaigi 2020/Sidebar/Views/SidebarViewController.swift]: Oh, I see. I will fix that!

**Comment by yaslab** [comment]
@roana0229 Thank you for your review! I have completed fix, please review again. 🙇‍♂️

**Comment by roana0229** [comment]
Thanks for quick work 👍

**Pull request merged**